src/shadowbox/infrastructure/prometheus_scraper.ts

```diff
@@ -70,11 +70,12 @@ export class PrometheusClient {
 }
 
 export function runPrometheusScraper(
-    args: string[], configFilename: string, configJson: {}): Promise<child_process.ChildProcess> {
+    args: string[], configFilename: string, configJson: {},
+    prometheusEndpoint: string): Promise<child_process.ChildProcess> {
   mkdirp.sync(path.dirname(configFilename));
   const ymlTxt = jsyaml.safeDump(configJson, {'sortKeys': true});
   return new Promise((resolve, reject) => {
-    fs.writeFile(configFilename, ymlTxt, 'utf-8', (err) => {
+    fs.writeFile(configFilename, ymlTxt, 'utf-8', async (err) => {
```

Comment:
This is fine, but I realized you can unnest the logic if you prefer (optional):
```ts
export async function runPrometheusScraper(..) {
...
// Async'fy writeFile
await new Promise((resolve, reject) => {
  fs.writeFile(configFilename, ymlTxt, 'utf-8', (err) => {
    if (err) { reject(err); }
    else { resolve(); }
  });
})
const commandArguments = ['--config.file', configFilename];
... // existing code here
runProcess.stderr.pipe(process.stderr);
await waitForPrometheusReady(prometheusEndpoint)
return process
```

Reply:
Good suggestion, I'm using `fs.writeFileSync` instead of waiting for the async version.